walkthrough/06-includes-vs-stacks/04-combining-patterns/README.md

```diff
@@ -0,0 +1,164 @@
+
+	parent_dir   = dirname(local.root_hcl_file)
+	project_name = read_terragrunt_config(local.root_hcl_file).locals.project_name
+
+	dist_dir   = "./dist"
+	source_dir = "./src"
+	
+	package_script = "./package.sh"
+
+	cur_dir_name = basename(get_terragrunt_dir())
+
+	environment_yml_path = find_in_parent_folders("environment.yml")
+
+	environment = yamldecode(file(local.environment_yml_path)).name
+}
+
+terraform {
+	before_hook "package" {
+		commands = ["plan", "apply"]
+		execute  = [local.package_script, local.source_dir, local.dist_dir]
+	}
+}
+
+inputs = {
+	name = "${local.project_name}-${local.cur_dir_name}-${local.environment}"
+
+	filename = "${local.dist_dir}/package.zip"
+}
+```
+
+```hcl
+# units/api/terragrunt.hcl
+include "root" {
+	path   = find_in_parent_folders("root.hcl")
+	expose = true
+}
+
+include "api" {
+	path = find_in_parent_folders("_envcommon/api.hcl")
+	merge_strategy = "deep"
+}
+
+locals {
+	parent_dir = dirname(find_in_parent_folders("root.hcl"))
+}
+
+terraform {
+	source = "${local.parent_dir}/modules/api"
+}
+
+dependency "db" {
+	config_path = "../db"
+
+	# Mock outputs allow us to continue to plan on the apply of the api module
+	# even though the db module has not yet been applied.
+	mock_outputs_allowed_terraform_commands = ["plan", "destroy"] 
+	mock_outputs = {
+		name = "mock-table"
+		arn  = "arn:aws:dynamodb:us-west-2:123456789012:table/mock-table"
+	}
+}
+
+inputs = {
+	dynamodb_table = dependency.db.outputs.name
+	dynamodb_arn   = dependency.db.outputs.arn
+}
+```
+
+```hcl
+# units/rng-api/terragrunt.hcl
+include "root" {
+	path   = find_in_parent_folders("root.hcl")
+	expose = true
+}
+
+include "api" {
+	path = find_in_parent_folders("_envcommon/api.hcl")
+	merge_strategy = "deep"
+}
+
+locals {
+	parent_dir = dirname(find_in_parent_folders("root.hcl"))
+}
+
+terraform {
+	source = "${local.parent_dir}/modules/rng-api"
+}
+```
+
+```hcl
+# live/dev/terragrunt.stack.hcl
+unit "api" {
+	source = "${dirname(find_in_parent_folders("root.hcl"))}/units/api"
+	path   = "api"
+}
+
+unit "rng_api" {
+	source = "${dirname(find_in_parent_folders("root.hcl"))}/units/rng-api"
+	path   = "rng-api"
+}
+
+unit "db" {
+	source = "${dirname(find_in_parent_folders("root.hcl"))}/units/db"
+	path   = "db"
+}
+```
+
+Here, you can see that both the `api` and `rng-api` units include the same `_envcommon/api.hcl` file, but the `api` unit also defines a `db` dependency. This allows the `api` unit to have the same base configuration as the `rng-api` unit, but also merge in configuration to use the `db` dependency. For teams that want a tight coupling between their unit configurations, this can be a good way to achieve that.
+
+Personally, I would actually prefer less-DRY configurations, and have `api` defined independently from `rng-api` with no `_envcommon`, but different teams have different needs. You have to decide where the correct trade-offs are for the infrastructure that you are managing.
```

Comment:
1) It's not clear from the document who "I" is, so I would leave out first person
2) I'd also leave out opinion
3) I think the fact that you have an opinion means that there's actually an implicit tradeoff.  For the sake of the walkthrough, it'd more effective to simply make the tradeoff explicit.